libs/recnet-api-model/src/lib/api/announcement.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from "zod";
+
+import { announcementSchema } from "../model";
+
+// GET /announcements
+export const getAnnouncementsParamsSchema = z.object({
+  page: z.coerce.number(),
+  pageSize: z.coerce.number(),
+  activatedOnly: z.coerce.boolean(),
+  currentOnly: z.coerce.boolean(),
```

Comment:
There's a catch. These query variables will be encoded as string in url. And `z.coerce.boolean()` will parse `"false"` as true (It uses constructor `Boolean()` under the hood). See the discussion in zod here: https://github.com/colinhacks/zod/issues/1630

Reply:
Workaround:

```ts
z.union([z.boolean(), z.literal("true"), z.literal("false")])
  .transform((value) => value === true || value === "true")
```

Reply:
Another issue, I think `activatedOnly` & `cuurentOnly` should be optional?

If so, sample workaround would be 
```ts
z.union([z.boolean(), z.literal("true"), z.literal("false")])
  .transform((value) => value === true || value === "true").optional()
  ```

Reply:
fixed